spec/support/authentication_helpers.rb

```diff
@@ -4,9 +4,17 @@
   # For user authentication
   config.before(:example, auth: :user_token) do
     user = Fabricate(:user)
-    access_token = Fabricate(:access_token, resource_owner_id: user.id)
+    application = Fabricate(:application)
+    access_token = Fabricate(:access_token, resource_owner_id: user.id, application_id: application.id)
 
     allow(controller).to receive(:doorkeeper_token) { access_token }
     request.headers['Content-Type'] = 'application/json'
   end
+
+  def oauth_application_params(application = Fabricate(:application))
+    {
+      client_id: application.uid,
+      client_secret: application.secret
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/10e383cbcb61a960ca8de7fb15a118e04aaac39f/spec/support/authentication_helpers.rb#L17"></span>Refers to ‘application’ more than self (maybe move it to another class?)</td>
    </tr>
  </tbody>
</table>